File: lib/features/saliks/domain/entities/bazam.dart
/// Default bazam assigned to seeded areas and new areas without an explicit id.
const kDefaultBazamId = 'i-10';

class Bazam {
  const Bazam({
    required this.bazamId,
    required this.bazamName,
  });

  final String bazamId;
  final String bazamName;

  factory Bazam.fromMap(Map<String, dynamic> map, {String? id}) {
    final bazamId = (map['bazamId'] as String?) ?? id;
    if (bazamId == null || bazamId.isEmpty) {
      throw const FormatException('Bazam document missing bazamId');
    }
    return Bazam(
      bazamId: bazamId,
      bazamName: (map['bazamName'] as String?)?.trim() ?? bazamId,
    );
  }

  Map<String, dynamic> toMap() => {
        'bazamId': bazamId,
        'bazamName': bazamName,
      };
}

File: app/src/main/java/com/example/salik_management_system/features/saliks/domain/Bazam.kt
package com.example.salik_management_system.features.saliks.domain

/** Default bazam assigned to seeded areas and new areas without an explicit id. */
const val kDefaultBazamId = "i-10"

data class Bazam(
    val bazamId: String,
    val bazamName: String,
) {
    fun toMap(): Map<String, Any?> = mapOf(
        "bazamId" to bazamId,
        "bazamName" to bazamName,
    )

    companion object {
        fun fromMap(map: Map<String, Any?>, id: String? = null): Bazam {
            val bazamId = (map["bazamId"] as? String) ?: id
            require(!bazamId.isNullOrEmpty()) { "Bazam document missing bazamId" }
            return Bazam(
                bazamId = bazamId,
                bazamName = (map["bazamName"] as? String)?.trim()?.ifEmpty { null } ?: bazamId,
            )
        }
    }
}
